### PR TITLE
biologicalEntitySet updated with draft WB fields

### PR DESCRIPTION
**Pull request by MagdalenaZZ**
Hi guys, 
I tried to add some slots to  GeneCluster,  GeneClass and  Operon, so that they will be able to harbour all WB data.
It is more of a draft than proper code, because I don't know exactly how to code it correctly, but I hope to learn :-)

### CODE REVIEW

File: model/schema/biologicalEntitySet.yaml [hunk 1]
```diff
@@ -56,18 +56,32 @@ classes:
     slots:
       - genes
 
+  GeneClass:
+    description: >-
+      A gene class is a set of genes which is of biological significance, and
```
Comment (sierra-moxon) [inline model/schema/biologicalEntitySet.yaml:61]: Is it possible to describe this in more detail - what does biological significance mean?  And can you give an example of the kinds of phenotype attributed to a group of genes?

File: model/schema/biologicalEntitySet.yaml [hunk 2]
```diff
@@ -56,18 +56,32 @@ classes:
     slots:
       - genes
 
+  GeneClass:
```
Comment (sierra-moxon) [inline model/schema/biologicalEntitySet.yaml:59]: GeneClass vs. GeneSet vs. GeneGroup ;)  - is this a GeneFamily?

File: model/schema/gene.yaml [hunk 3]
```diff
@@ -86,3 +94,31 @@ slots:
     description: SOTerm describing gene type
     domain: Gene
     range: SOTerm
+
+  gene_designating_laboratory:
```
Comment (markquintontulloch) [inline model/schema/gene.yaml:98]: I assume the intention of this slot is the same for Gene and GeneClass classes, so you can just use 'designating_laboratory' as defined in biologicalEntitySet.yaml and remove the `domain: GeneClass` line from the slot definition.  You can then either import biologicalEntitySet.yaml in gene.yaml or move the slot definition to core.yaml

File: model/schema/gene.yaml [hunk 4]
```diff
@@ -86,3 +94,31 @@ slots:
...
+    multivalued: false
+    domain: GeneClass
+    range: Laboratory
+
+  designating_person:
+    description: >-
+      A person/persons who designated this gene
+    required: false
+    multivalued: true
+    domain: GeneClass
```
Comment (markquintontulloch) [inline model/schema/gene.yaml:111]: This should be Gene

File: model/schema/gene.yaml [hunk 5]
```diff
@@ -86,3 +94,31 @@ slots:
...
+    range: Laboratory
+
+  splice_leader:
+    description: >-
+      Trans-splicing splice leaders observed to operate on this gene 
+    required: false
+    multivalued: true
+    domain: GeneClass
```
Comment (markquintontulloch) [inline model/schema/gene.yaml:119]: Also should be Gene

File: model/schema/gene.yaml [hunk 6]
```diff
@@ -86,3 +94,31 @@ slots:
...
+
+  gene_designating_laboratory:
+    description: >-
+      A laboratory which designated this gene
+    required: false
+    multivalued: false
+    domain: GeneClass
+    range: Laboratory
+
+  designating_person:
```
Comment (markquintontulloch) [inline model/schema/gene.yaml:106]: For multivalue slots we tend to pluralise the slot name, i.e. designating_persons

Comment (MagdalenaZZ) [reply model/schema/gene.yaml:117]: done

File: model/schema/gene.yaml (hunk 4 above)
Comment (MagdalenaZZ) [reply model/schema/gene.yaml]: done

File: model/schema/biologicalEntitySet.yaml (hunk 1 above)
Comment (MagdalenaZZ) [reply model/schema/biologicalEntitySet.yaml]: description improved

File: model/schema/gene.yaml (hunk 3 above)
Comment (markquintontulloch) [reply model/schema/gene.yaml:109]: It's perfectly valid LinkML, but if the slots are intended to capture the same information then I don't see a need for duplication of code

File: model/schema/gene.yaml [hunk 7]
```diff
@@ -86,3 +105,107 @@ slots:
...
+
+  wbprocess:
+    description: >-
+      WB specific. Eg WBbiopr:00000015 (Corpse engulfment) 
+    required: false
+    multivalued: true
+    domain: Gene
+    range: string  # this should be changed to a WBbiopr term
+
+  genetic_map_chromosome:
```
Comment (markquintontulloch) [inline model/schema/gene.yaml:173]: Might be worth considering creating a GeneticMapPosition class?

Comment (markquintontulloch) [reply model/schema/gene.yaml:164]: I'm not seeing any changes?

File: model/schema/gene.yaml (hunk 3 above)
Comment (markquintontulloch) [reply model/schema/gene.yaml:108]: 'fixed' or 'decided against' - I don't see any changes?

File: model/schema/gene.yaml (hunk 7 above)
Comment (markquintontulloch) [reply model/schema/gene.yaml:197]: Now I see them in the latest commit :)

**Comment by MagdalenaZZ** [comment]
ok - seems like I could do it myself. Happy days :-)

**Pull request merged**